src/services/dungeonService.js

```diff
@@ -182,6 +182,33 @@ class DungeonService {
       });
     });
   }
+
+  async updateScoutedObstacles(heroId, time) {
+    const oldScoutedObstacles = await this.scoutedInstance(heroId);
+    const oldTime = await this.heroService.getIdleTimestamp(heroId);
+    let query = '';
+    return new Promise((resolve, reject) => {
+      if (time >= 1) {
+        query = `UPDATE dungeoninstance SET scoutedObstacles = ${oldScoutedObstacles + time} WHERE (heroId = ?);`;

```

Comment:
If oldScoutedObstacles or time came as a parameter, this would be still a possibility for sql injection!